libs/ui-lib/lib/ocm/components/clusterDetail/AssistedInstallerDetailCard.tsx

```diff
@@ -145,35 +145,45 @@ const AssistedInstallerDetailCard = ({
 
   const isOutdatedClusterData = uiState === ResourceUIState.POLLING_ERROR;
 
-  return (
-    <HistoryRouter history={history} basename={basename}>
-      <AlertsContextProvider>
-        <OpenshiftVersionsContextProvider>
-          <SentryErrorMonitorContextProvider>
-            <ModalDialogsContextProvider>
-              <ClusterDefaultConfigurationProvider
-                loadingUI={<LoadingCard />}
-                errorUI={<LoadingDefaultConfigFailedCard />}
+  const clusterDetailCard = (
+    <AlertsContextProvider>
+      <OpenshiftVersionsContextProvider>
+        <SentryErrorMonitorContextProvider>
+          <ModalDialogsContextProvider>
+            <ClusterDefaultConfigurationProvider
+              loadingUI={<LoadingCard />}
+              errorUI={<LoadingDefaultConfigFailedCard />}
+            >
+              <NewFeatureSupportLevelProvider
+                loadingUi={<LoadingCard />}
+                cluster={cluster}
+                cpuArchitecture={infraEnv.cpuArchitecture as CpuArchitecture}
+                openshiftVersion={cluster.openshiftVersion}
+                platformType={cluster.platform?.type}
               >
-                <NewFeatureSupportLevelProvider
-                  loadingUi={<LoadingCard />}
-                  cluster={cluster}
-                  cpuArchitecture={infraEnv.cpuArchitecture as CpuArchitecture}
-                  openshiftVersion={cluster.openshiftVersion}
-                  platformType={cluster.platform?.type}
-                >
-                  {content}
-                </NewFeatureSupportLevelProvider>
-                {isOutdatedClusterData && <ClusterPollingErrorModal />}
-                <CancelInstallationModal />
-                <ResetClusterModal />
-                <DiscoveryImageModal />
-              </ClusterDefaultConfigurationProvider>
-            </ModalDialogsContextProvider>
-          </SentryErrorMonitorContextProvider>
-        </OpenshiftVersionsContextProvider>
-      </AlertsContextProvider>
-    </HistoryRouter>
+                {content}
+              </NewFeatureSupportLevelProvider>
+              {isOutdatedClusterData && <ClusterPollingErrorModal />}
+              <CancelInstallationModal />
+              <ResetClusterModal />
+              <DiscoveryImageModal />
+            </ClusterDefaultConfigurationProvider>
+          </ModalDialogsContextProvider>
+        </SentryErrorMonitorContextProvider>
+      </OpenshiftVersionsContextProvider>
+    </AlertsContextProvider>
+  );
+
+  return (
+    <>
+      {history && history.action ? (
+        <HistoryRouter history={history} basename={basename}>
```

Comment:
Basename could potentially be `undefined`. In [Routes.tsx](https://github.com/openshift-assisted/assisted-installer-ui/blob/master/libs/ui-lib/lib/ocm/components/Routes.tsx#L46) we supply a fallback value. Do we need to do the same here or not?

Reply:
changed it